nativelink-store/src/filesystem_store.rs

```diff
@@ -848,9 +867,16 @@ impl<Fe: FileEntry> StoreDriver for FilesystemStore<Fe> {
             return Ok(());
         }
 
-        let entry =
-            self.evicting_map.get(&digest).await.ok_or_else(|| {
-                make_err!(Code::NotFound, "{digest} not found in filesystem store")
+        let entry = self
```

Comment:
How have we evaluated whether or not we have increased heap allocation overhead? I'm not sure if you used a tool like Valgrind or something similar. Could you elaborate? I'm not saying that there is a problem here. I'm just saying that it's possible. 

I will also try to find some time this week to look into it, but this one is important to complete the task because we will be adding this code to a large scale system.

Reply:
I can actually just remove this  `into_owned()` using the method described here https://blinsay.com/blog/compound-keys/ however as the wrapper and inner structs are reversed (cannot easily make the currently existing StoreKey a wrapper struct around the actual enum, and cannot easily make everywhere storing StoreKey instead use StoreKeyWrapper<'static>, instead use unsafe to transmute the reference between the two, ensuring the wrapper is transparent so they are guaranteed to have the same layout in memory - I have not yet squashed commits so if wanted, this can be easily reverted

Reply:
I want more co-founder @allada and @aaronmondal to take a look before we adopt this change. @allada is out until Monday but he might get a chance to review this change before we then. I appreciate that we have both looks so no need to squash just yet. 

Thanks again. Great work thus far!

Reply:
This does acc also work to remove the todo on the has_with_results for filesystem_store.rs and memory_store.rs but will push on top so easy reverts still. No worries if it takes a bit!

Reply:
Have a safe version now too, so should be easier to review!